src/ImageSharp/Formats/WebP/AlphaDecoder.cs

```diff
@@ -0,0 +1,309 @@
+        public void AlphaApplyFilter(int firstRow, int lastRow, Span<byte> dst, int stride)
+        {
+            if (this.AlphaFilterType is WebPAlphaFilterType.None)
+            {
+                return;
+            }
+
+            Span<byte> alphaSpan = this.Alpha.Memory.Span;
+            Span<byte> prev = this.PrevRow == 0 ? null : alphaSpan.Slice(this.Width * this.PrevRow);
+            for (int y = firstRow; y < lastRow; ++y)
+            {
+                switch (this.AlphaFilterType)
+                {
+                    case WebPAlphaFilterType.Horizontal:
+                        HorizontalUnfilter(prev, dst, dst, this.Width);
+                        break;
+                    case WebPAlphaFilterType.Vertical:
+                        VerticalUnfilter(prev, dst, dst, this.Width);
+                        break;
+                    case WebPAlphaFilterType.Gradient:
+                        GradientUnfilter(prev, dst, dst, this.Width);
+                        break;
+                }
+
+                prev = dst;
+                dst = dst.Slice(stride);
+            }
+
+            this.PrevRow = lastRow - 1;
+        }
+
+        private static void HorizontalUnfilter(Span<byte> prev, Span<byte> input, Span<byte> dst, int width)
+        {
+            byte pred = (byte)(prev == null ? 0 : prev[0]);
+
+            for (int i = 0; i < width; ++i)
+            {
+                dst[i] = (byte)(pred + input[i]);
+                pred = dst[i];
+            }
+        }
+
+        private static void VerticalUnfilter(Span<byte> prev, Span<byte> input, Span<byte> dst, int width)
+        {
+            if (prev == null)
+            {
+                HorizontalUnfilter(null, input, dst, width);
+            }
+            else
+            {
+                for (int i = 0; i < width; ++i)
+                {
+                    dst[i] = (byte)(prev[i] + input[i]);
+                }
+            }
+        }
+
+        private static void GradientUnfilter(Span<byte> prev, Span<byte> input, Span<byte> dst, int width)
+        {
+            if (prev == null)
+            {
+                HorizontalUnfilter(null, input, dst, width);
+            }
+            else
+            {
+                byte top = prev[0];
+                byte topLeft = top;
+                byte left = top;
+                for (int i = 0; i < width; ++i)
+                {
+                    top = prev[i];
+                    left = (byte)(input[i] + GradientPredictor(left, top, topLeft));
+                    topLeft = top;
+                    dst[i] = left;
+                }
+            }
+        }
+
+        private static bool Is8bOptimizable(Vp8LMetadata hdr)
+        {
+            if (hdr.ColorCacheSize > 0)
+            {
+                return false;
+            }
+
+            // When the Huffman tree contains only one symbol, we can skip the
+            // call to ReadSymbol() for red/blue/alpha channels.
```

Comment:
@jongleur1983 im keen in removing Is8bOptimizable altogether, because its not used yet and i dont see when we would need more than 1 byte per pixel for the alpha channel. I have no test image that would need that. Do you have any insight were this might be the case? 

I dont see anything mentioned in the [spec](https://developers.google.com/speed/webp/docs/riff_container#alpha), but im afraid that it must have some use case im not aware of.

Reply:
i have found one image where Is8bOptimizable is false: `alpha_color_cache.webp`
I will try to figure out what the difference is to the other images.

Reply:
i have figured out how to decode `alpha_color_cache.webp`. The alpha channel data will be in the green channel of the decoded lossless data. See commit: 45bf1064e34a575d77ddbfafc6d7ffb935d172d7